ton_client/src/debot/calltype.rs

```diff
@@ -161,8 +167,14 @@ async fn decode_and_fix_ext_msg(
 
     let mut new_body = BuilderData::new();
     let pubkey = signer.resolve_public_key(ton.clone()).await?;
-    if meta.is_pubkey {
-        if let Some(ref key) = pubkey {
+    if meta.is_pubkey ||  meta.ex_provided_pubkey.is_some() {
```

Comment:
is it necessary to check ex_provided_pubkey here? let's assume that is `is_pubkey` is set then we should choose between `ex_provided_pubkey` or `pubkey`.